org.ultimatepp.TheIDE.yml

```diff
@@ -0,0 +1,113 @@
+app-id: org.ultimatepp.TheIDE
+runtime: org.gnome.Platform
+runtime-version: '45'
+sdk: org.gnome.Sdk
+sdk-extensions: ["org.freedesktop.Sdk.Extension.llvm17"]
+command: run-ide.sh
+finish-args:
+  - --socket=x11
+  - --share=network
+  - --share=ipc
+  - --filesystem=host
+  - --talk-name=org.freedesktop.Flatpak
+modules:
+  - name: upp
+    buildsystem: simple
+    build-commands: 
+      - printf "#define flagFlatpak 1\\n" > uppsrc/uppconfig.h
+      - make -f Makefile -j $FLATPAK_BUILDER_N_JOBS
+      - mv ide theide
+      - install -Dm755 theide /app/bin/theide
+      - mkdir /app/src
+      - mkdir /app/lib
+      - printf "\\n" > uppsrc/uppconfig.h
+      - rm -f /app/src/ver.txt && touch /app/src/ver.txt
+      - printf "2023.2" > /app/src/ver.txt
+      - cp -r examples /app/src
+      - cp -r reference /app/src
+      - cp -r tutorial /app/src
+      - cp -r uppsrc /app/src
+      - cp .clang-format /app/src/uppsrc/.clang-format
+      - cp run-ide-install-host-deps.sh /app/src/run-ide-install-host-deps.sh
+      - cp -r /usr/lib/sdk/llvm17/lib /app
+    sources:
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/archive/refs/tags/2023.2.tar.gz
+        sha256: 8e0f02d2cdcd23f7774e9c576a8cc7afa50b97232046d09940e386cb89dcf286
+      - type: file
+        path: run-ide-install-host-deps.sh
+      - type: patch
+        path: flatpak.patch
+  
+  - name: upp-term
+    buildsystem: simple
+    build-commands:
+      - mv 3p/Terminal/PtyProcess app/PtyProcess
+      - mv 3p/Terminal/PtyAgent app/PtyAgent
+      - mv 3p/Terminal/Terminal app/Terminal
+      - 3p/umk/umk app/,3p/uppsrc UppTerm 3p/umk/GCC.bm -brv +GUI,SHARED upp-term
+      - install -Dm755 upp-term /app/bin/upp-term
+    sources:
+      - type: archive
+        url: https://github.com/ultimatepp/UppTerm/archive/refs/tags/v2023.1.tar.gz
+        sha256: a40da7148d16f8f7c9989da1ad430d69c09483c91dab0607bc422ae6c184dda2
+      - type: archive
+        url: https://github.com/ismail-yilmaz/Terminal/archive/refs/tags/2023.2.tar.gz
+        sha256: 320d8574842825d1d8ddfa46addcb4eb03a9120116caea192582f2a02d26ec09
+        dest: 3p/Terminal
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/releases/download/2023.2/uppsrc-17045.tar.gz
+        sha256: 23793a363a140139728a30cc050c35287d6ff97ee03125d62f8ad8acf06e40af
+        dest: 3p/uppsrc
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/releases/download/2023.2/umk-17045-linux-x86-64.tar.gz
```

Comment:
can't this be built from source?

Reply:
This is probably doable. I will need to do investigation and I will back to you.

Reply:
Done!